src/main/java/com/aws/greengrass/deployment/DeploymentConfigMerger.java

```diff
@@ -106,7 +108,11 @@ public Future<DeploymentResult> mergeInNewConfig(Deployment deployment,
         } else {
             logger.atInfo().log("Deployment is configured to skip update policy check,"
                     + " not waiting for disruptable time to update");
-            updateActionForDeployment(newConfig, deployment, activator, totallyCompleteFuture);
+            // use executor service to execute updateActionForDeployment
+            // prevents default deployment cancellation from directly interrupting
```

Comment:
Just for my understanding, where's the interrupt happening?

Reply:
mergeInNewConfig is called here:
https://github.com/aws-greengrass/aws-greengrass-nucleus/blob/main/src/main/java/com/aws/greengrass/deployment/DefaultDeploymentTask.java#L154

and the interruption is caught just below that:
https://github.com/aws-greengrass/aws-greengrass-nucleus/blob/main/src/main/java/com/aws/greengrass/deployment/DefaultDeploymentTask.java#L178

If it's not running in a different thread, it will be interrupted when we try to cancel a deployment here:
https://github.com/aws-greengrass/aws-greengrass-nucleus/blob/main/src/main/java/com/aws/greengrass/deployment/DeploymentService.java#L500

Reply:
`updateActionForDeployment` doesn't throw interrupted exception though?

also I'm not entirely clear on the use case, what's the behavior that we're trying to prevent? e.g. deployments aren't getting cancelled properly?

Reply:
updateActionForDeployment doesn't throw interrupted exception no, but currently it's running on the same thread as the Default Deployment task, which means it will be interrupted when we cancel the deployment Future.

For reference `resolveDependencies` and `preparePackages` both execute in their own thread.

- https://github.com/aws-greengrass/aws-greengrass-nucleus/blob/main/src/main/java/com/aws/greengrass/deployment/DefaultDeploymentTask.java#L124
- https://github.com/aws-greengrass/aws-greengrass-nucleus/blob/main/src/main/java/com/aws/greengrass/componentmanager/ComponentManager.java#L350

When we begin the config merge we don't want it to be interrupted, instead we want to cleanly exit as soon as possible by checking if the CompletableFuture is cancelled or not.